src/Utils.ts

```diff
@@ -23,18 +23,16 @@ export function fromEditorConfig(
 	if (get(resolved, 'tabSize') === 'tab') {
 		resolved.tabSize = config.tab_width;
 	}
-	if (config.indent_style !== 'tab'
-		&& config.indent_style !== 'space'
-		&& config.indent_size !== 'tab') {
-			return {
-				tabSize: get(resolved, 'tabSize', defaults.tabSize)
-			};
-	}
 	return {
-		insertSpaces: config.indent_style
-			? config.indent_style !== 'tab'
-			: defaults.insertSpaces,
-		tabSize: get(resolved, 'tabSize', defaults.tabSize)
+		...(config.indent_style === 'tab'
+			|| config.indent_size === 'tab'
+			|| config.indent_style === 'space'
+		) ? {
+			insertSpaces: config.indent_style === 'space'
+		} : {},
+		tabSize: resolved.tabSize >= 0
```

Comment:
Are we saying people can't have a `tab_width` of `0` now? I realize it doesn't make much sense to have a `tab_width` of `0`, but technically it was supported before.

Reply:
oh no a zero will be fine still, only that we defer to workspace config if we only have tab_width on its own. Oddly, the spec of editorconfig is conflict with VSCode's definition of tabSize! 
we can be correct for either:
```
test/fixtures/detect-indentation/tab_width/indent-style-space
test/fixtures/detect-indentation/tab_width/indent-style-tab
```
but not both :(

Reply:
[edit]:
in terms of desirable behaviour 
test/fixtures/detect-indentation/tab_width/indent-style-space [detect spaces of 2]
test/fixtures/detect-indentation/tab_width/indent-style-tab [use tab size of 8 from tab_width]

Reply:
Can you elaborate on what exactly is conflicting? I'm not understanding that point.

Reply:
If I have `tab_width = 0` in my .editorconfig file and I get to this condition where it says `resolved.tabSize >= 0 ? resolved.tabSize : defaults.tabSize` then that means it would use the `defaults.tabSize` instead of the `tab_width = 0` that I had indicated. Am I missing something here in this scenario?